### PR TITLE
Fix AOTAutograd 2.0 perf regression involving as_strided

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92255

I feel there may be a deeper fix where we avoid as_strided entirely, but in the regressed model the sizes/strides all lined up exactly, so this seems to work to fix the immediate regression.

Repro command: `python benchmarks/dynamo/torchbench.py  --performance  --backend inductor --float16 --training --batch-size-file $(realpath benchmarks/dynamo/torchbench_models_list.txt) --only hf_Bert  `

Before: 1.138x p=0.00
After: 1.162x p=0.00 

Natalia pinpointed it to this line by comparing GPU traces and finding that the regressed PyTorch had two extra fill kernels and a memcpy:

Without regression: 
image

With regression:
image

...which CPU profiler blamed on `AsStridedBackward`:

image

...which were then pinpointed to  https://github.com/pytorch/pytorch/pull/92076/files#diff-df954bbf954d2dcb81f687876053267ffa4ddb36ed86b7d2bd76319ff2b94416R486-R489

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

File: torch/_functorch/aot_autograd.py [hunk 1]
```diff
@@ -483,9 +483,15 @@ def gen_alias_from_base(aliased_base_tensor, target_meta_tensor, target_requires
     if target_meta_tensor._base is not None:
         # The base that we want to replay our view off of might have a different shape than the view's original base.
         b = target_meta_tensor._base
-        reshaped_base_tensor = aliased_base_tensor.as_strided(
-            b.size(), b.stride(), b.storage_offset()
-        )
+        abt = aliased_base_tensor
+        # Don't unnecessarily call as_strided if nothing changed; as_strided's
+        # backward is poorly implemented and slow
+        if abt.size() != b.size() or abt.stride() != b.stride() or abt.storage_offset() != b.storage_offset():
```
Comment (ngimel) [inline torch/_functorch/aot_autograd.py:489]: not that it matters much, but a faster check would do in most cases: `if abt is not b or ...:

**Review by bdhirsh — APPROVED** [review]
oof. Thanks! (and thanks for providing the repro command + debugging steps)

**Pull request closed without merging**